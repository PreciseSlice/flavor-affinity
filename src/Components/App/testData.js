/* eslint-disable */
export const allIngredients = [
  {
    id: 5902,
    name: '1724 Tonic',
    product: '1724 Tonic',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 4578,
        src: 'https://pantry.foodpairing.com/apps/4578',
        size_240: 'https://pantry.foodpairing.com/apps/4578_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/4578_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/4578_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/4578_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/4578_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/4578_1440_0'
      },
      brand: null,
      categories: [
        {
          id: 38,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/38'
        },
        {
          id: 139,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/139'
        }
      ]
    },
    description:
      'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 0
    }
  },
  {
    id: 5440,
    name: 'A la Chimay Bleue',
    product: 'A la Chimay Bleue',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1772,
        src: 'https://pantry.foodpairing.com/apps/1772',
        size_240: 'https://pantry.foodpairing.com/apps/1772_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1772_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1772_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1772_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1772_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1772_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 1
    }
  },
  {
    id: 5443,
    name: 'A la Chimay Rouge',
    product: 'A la Chimay Rouge',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1773,
        src: 'https://pantry.foodpairing.com/apps/1773',
        size_240: 'https://pantry.foodpairing.com/apps/1773_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1773_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1773_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1773_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1773_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1773_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 2
    }
  }
];

export const cleanData = [
    {
      id: 5902,
      name: '1724 tonic',
      image: 'https://pantry.foodpairing.com/apps/4578_240_0',
      description:
        'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.'
    },
    {
      id: 5440,
      name: 'a la chimay bleue',
      image: 'https://pantry.foodpairing.com/apps/1772_240_0',
      description:
        'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.'
    },
    {
      id: 5443,
      name: 'a la chimay rouge',
      image: 'https://pantry.foodpairing.com/apps/1773_240_0',
      description:
        'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.'
    }
];
