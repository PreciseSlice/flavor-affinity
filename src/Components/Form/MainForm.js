import React, {Component} from 'react';
import './Form.css';

class MainForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userSearch: ''
    } 
  }

  handleChange(event) {
    const {name, value} = event.target;
    this.setState({ [name]: value });
  }

  render() {
    return (
      <div className="form-container">
        <form>
          <input 
            name="userSearch" 
            type="text" 
            onChange={ event => this.handleChange(event) } 
            placeholder="enter ingredient" />
          {/* <buttton>img</buttton> */}
        </form>
      </div>
    )
  }
}

export default MainForm;