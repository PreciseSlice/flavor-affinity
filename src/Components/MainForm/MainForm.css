input {
  height: 30px;
  width: 300px;
  border-radius: 30px;
  outline: none; }

::-webkit-input-placeholder {
  text-align: center; }

:-moz-placeholder {
  /* Firefox 18- */
  text-align: center; }

::-moz-placeholder {
  /* Firefox 19+ */
  text-align: center; }

:-ms-input-placeholder {
  text-align: center; }

button {
  border-radius: 30px;
  height: 33px;
  width: 90px;
  margin-top: 30px;
  outline: none; }

.form-container {
  margin-top: 30px; }
