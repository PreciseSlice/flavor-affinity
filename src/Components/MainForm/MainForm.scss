input {
  height: 30px;
  width: 300px;
  border-radius: 30px;
}

::-webkit-input-placeholder {
  text-align: center;
}

:-moz-placeholder { 
  /* Firefox 18- */
  text-align: center;  
}

::-moz-placeholder {  
  /* Firefox 19+ */
  text-align: center;  
}

:-ms-input-placeholder {  
  text-align: center; 
}

button {
  border-radius: 30px;
  height: 33px;
  width: 90px;
  margin-left: 10px;
}