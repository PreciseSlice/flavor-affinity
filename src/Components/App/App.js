import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import MainForm from '../MainForm/MainForm';
import CardContainer from '../CardContainer/CardContainer';
//import { getAllIngredients } from '../../Helpers/apiCalls';
import { setAllIngredients } from '../../Actions';
//import * as actions from '../../Actions';
import { allDataClean } from '../../Helpers/.mockApi.js';
//import { cleanData } from '../App/testData';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      errorInfo: null
    };
  }

  async componentDidMount() {
    const { setAllIngredients } = this.props;
    //const allIngredients = await getAllIngredients();
    //setAllIngredients(allIngredients);
    setAllIngredients(allDataClean);
  }

  componentDidCatch(error, errorInfo) {
    this.setState({
      error: error,
      errorInfo: errorInfo
    });
  }

  clearError() {
    this.setState({
      error: null,
      errorInfo: null
    });
  }

  render() {
    if (this.state.error) {
      return (
        <div className="eror">
          <h2>{'Oh-no! Something went wrong'}</h2>
          <p className="">{this.state.error && this.state.error.toString()}</p>
          <div>{'Component Stack Error Details: '}</div>
          <p className="">{this.state.errorInfo.componentStack}</p>
          <button onClick={() => this.clearError()} />
        </div>
      );
    }
    return (
      <div className="app">
        <MainForm />
        <CardContainer />
      </div>
    );
  }
}

// export const mapStateToProps = state => ({

// });

export const mapDispatchToProps = dispatch => ({
  setAllIngredients: ingredient => dispatch(setAllIngredients(ingredient))
});

export default connect(null, mapDispatchToProps)(App);
