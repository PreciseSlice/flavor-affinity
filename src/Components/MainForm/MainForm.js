import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  setAllIngredients,
  setSuggestedIngredients
} from '../../Actions/index';
//import * as actions from '../../Actions';
//import { searchForIngredient } from '../../Helpers/apiCalls';
import { PrefixTrie } from '@PreciseSlice/complete-me';
import './MainForm.css';
import PropTypes from 'prop-types';

export class MainForm extends Component {
  constructor(props) {
    super(props);

    this.trie = new PrefixTrie();

    this.state = {
      userInput: ''
    };
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.allIngredients !== nextProps.allIngredients) {
      const { allIngredients } = nextProps;
      const nameArray = allIngredients.map(ingredient => ingredient.name);

      this.trie.populate(nameArray);
    }
  }

  handleChange(event) {
    const { name, value } = event.target;

    this.setState({ [name]: value });
    this.suggestIngredient(value);
  }

  // async submitForm(event) {
  //   event.preventDefault();
  //   const { setAllIngredients } = this.props;
  //   const { userInput } = this.state;
  //   const searchReasult = await searchForIngredient(userInput);
  //   setAllIngredients(searchReasult);
  // }

  suggestIngredient(input) {
    const suggestedIngredients = this.trie.suggest(input);

    this.props.setSuggestedIngredients(suggestedIngredients);
  }

  render() {
    return (
      <div className="form-container">
        <form /* onSubmit={event => this.submitForm(event) } */>
          <input
            autoComplete="off"
            autoFocus
            name="userInput"
            type="text"
            onChange={event => this.handleChange(event)}
            placeholder="enter ingredient"
            list="drop-down"
          />

          <datalist id="drop-down">
            {this.props.suggestedIngredients
              .map((ingredient, i) => {
                return <option value={ingredient} key={i} />;
              })
              .slice(0, 5)}
          </datalist>

          <button /* onClick={event => this.submitForm(event)} */>
            search
          </button>
        </form>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  allIngredients: state.ingredients,
  suggestedIngredients: state.suggestedIngredients
});

export const mapDispatchToProps = dispatch => ({
  setAllIngredients: ingredient => dispatch(setAllIngredients(ingredient)),
  setSuggestedIngredients: suggestedIngredients =>
    dispatch(setSuggestedIngredients(suggestedIngredients))
});

export default connect(mapStateToProps, mapDispatchToProps)(MainForm);

MainForm.propTypes = {
  allIngredients: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.arrayOf(
      PropTypes.shape({
        description: PropTypes.string,
        id: PropTypes.number.isRequired,
        image: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })
    )
  ])
};
