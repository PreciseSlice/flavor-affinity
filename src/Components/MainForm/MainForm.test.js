/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { MainForm, mapDispatchToProps, mapStateToProps } from './MainForm';
import { shallow } from 'enzyme';
import { cleanData } from '../App/testData';

describe('MainForm', () => {
  let wrapper;
  const mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <MainForm
        suggestedIngredients={cleanData}
        setSuggestedIngredients={mockFn}
        setUserInput={mockFn}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentWillReciveProps', () => {
    it('should call tries populate method once props are recieved', () => {
      const mockFn = jest.fn();
      const spy = jest.spyOn(MainForm.prototype, 'componentWillReceiveProps');
      const wrapper = shallow(
        <MainForm
          suggestedIngredients={cleanData}
          setSuggestedIngredients={mockFn}
          setUserInput={mockFn}
        />
      );

      expect(
        MainForm.prototype.componentWillReceiveProps
      ).not.toHaveBeenCalled();

      wrapper.setProps({ suggestedIngredients: cleanData });

      expect(MainForm.prototype.componentWillReceiveProps).toHaveBeenCalled();

      // need to mock this
      //expect(wrapper.instance().trie.populate).toHaveBeenCalled()

      spy.mockReset();
      spy.mockRestore();
    });
  });

  describe('handleChange', () => {
    const event = { target: { value: 'Apple', name: 'userInput' } };

    it.skip('should set the user input to state', () => {
      expect(wrapper.state().userInput).toEqual('');
      wrapper.instance().handleChange(event);
      expect(wrapper.state().userInput).toEqual('Apple');
    });

    it.skip('should call suggestIngredient', () => {
      const suggestIngredient = jest.fn();
      wrapper.instance().handleChange(event);
      expect(wrapper.instance().suggestIngredient).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should map items in the store to props', () => {
      const mockStore = {
        state: {
          ingredients: cleanData,
          suggestedIngredients: cleanData
        }
      };
      const mapped = mapStateToProps(mockStore);

      expect(mapped.allIngredients).toEqual(mockStore.ingredients);
      expect(mapped.suggestedIngredients).toEqual(
        mockStore.suggestedIngredients
      );
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call the dispatch function when using a function from mapDispachToProps', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);
      //verify what is being returned from MDTP object
      // .toHaveBeenCalledWith(expectedParams)
      mapped.setAllIngredients();
      mapped.setSuggestedIngredients();

      expect(mockDispatch).toHaveBeenCalledTimes(2);
    });
  });
});
