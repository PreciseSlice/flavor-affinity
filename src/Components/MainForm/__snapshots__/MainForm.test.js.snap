// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainForm Exist and matches snapshot 1`] = `
<div
  className="form-container"
>
  <form>
    <input
      autoComplete="off"
      autoFocus={true}
      list="drop-down"
      name="userInput"
      onChange={[Function]}
      placeholder="enter ingredient"
      type="text"
    />
    <datalist
      id="drop-down"
    >
      <option
        key="0"
        value={
          Object {
            "description": "The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.",
            "id": 5902,
            "image": "https://pantry.foodpairing.com/apps/4578_240_0",
            "name": "1724 tonic",
          }
        }
      />
      <option
        key="1"
        value={
          Object {
            "description": "The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.",
            "id": 5440,
            "image": "https://pantry.foodpairing.com/apps/1772_240_0",
            "name": "a la chimay bleue",
          }
        }
      />
      <option
        key="2"
        value={
          Object {
            "description": "The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.",
            "id": 5443,
            "image": "https://pantry.foodpairing.com/apps/1773_240_0",
            "name": "a la chimay rouge",
          }
        }
      />
    </datalist>
    <button>
      search
    </button>
  </form>
</div>
`;
